src/components/mgt-get/mgt-get.ts

```diff
@@ -0,0 +1,192 @@
+    }
+  })
+  public scopes: string[] = [];
+
+  /**
+   * Api version to use for request
+   *
+   * @type {string}
+   * @memberof MgtGet
+   */
+  @property({
+    attribute: 'version',
+    type: String
+  })
+  public version: string = 'v1.0';
+
+  /**
+   * Maximum number of pages to get for the resource
+   * default = 3
+   * if <= 0, all pages will be fetched
+   *
+   * @type {boolean}
+   * @memberof MgtGet
+   */
+  @property({
+    attribute: 'max-pages',
+    type: Number
+  })
+  public maxPages: number = 3;
+
+  /**
+   * Gets or sets the response of the request
+   *
+   * @type {*}
+   * @memberof MgtGet
+   */
+  @property({ attribute: false }) public response: any;
+
+  /**
+   *
+   * Gets or sets the error (if any) of the request
+   * @type {*}
+   * @memberof MgtGet
+   */
+  @property({ attribute: false }) public error: any;
+
+  @property({ attribute: false }) private loading: boolean = false;
+  private _firstUpdated: boolean = false;
+
+  /**
+   * Synchronizes property values when attributes change.
+   *
+   * @param {*} name
+   * @param {*} oldValue
+   * @param {*} newValue
+   * @memberof MgtPersonCard
+   */
+  public attributeChangedCallback(name, oldval, newval) {
+    super.attributeChangedCallback(name, oldval, newval);
+
+    if (this._firstUpdated) {
+      this.loadData();
+    }
+  }
+
+  /**
+   * Invoked when the element is first updated. Implement to perform one time
+   * work on the element after update.
+   *
+   * Setting properties inside this method will trigger the element to update
+   * again after this update cycle completes.
+   *
+   * * @param _changedProperties Map of changed properties with old values
+   */
+  public firstUpdated() {
+    Providers.onProviderUpdated(() => this.loadData());
+    this.loadData();
+    this._firstUpdated = true;
+  }
+
+  /**
+   * Invoked on each update to perform rendering tasks. This method must return
+   * a lit-html TemplateResult. Setting properties inside this method will *not*
+   * trigger the element to update.
+   */
+  protected render() {
+    if (this.loading) {
+      return this.renderTemplate('loading', null);
+    } else if (this.error) {
+      return this.renderTemplate('error', this.error);
+    } else if (this.response) {
+      return this.renderTemplate('default', this.response);
+    }
+  }
+
+  private async loadData() {
+    const provider = Providers.globalProvider;
+
+    if (!provider || provider.state !== ProviderState.SignedIn) {
+      return;
+    }
+
+    this.response = null;
+    this.error = null;
+
+    if (this.resource) {
+      this.loading = true;
+
+      let response = null;
+      try {
+        let request = provider.graph.client.api(this.resource).version(this.version);
+
+        if (this.scopes && this.scopes.length) {
+          request = request.middlewareOptions(prepScopes(...this.scopes));
+        }
+
+        response = await request.get();
```

Comment:
Will the UI update at this point when this is set or wait until the rest of the pages are fetched?

It'd be nice if the control could start loading at this point and show the first set of items while the rest are fetched?

Thinking from the XAML side of just having an Observable Collection, so items will be added in batches as the data is fetched and the UI updates as they come in?